aten/src/ATen/native/cuda/GridSampler.cu

```diff
@@ -754,9 +757,10 @@ void launch_grid_sampler_2d_forward_kernel(
   int64_t count = N * H * W;
   if (count > 0) {
     AT_DISPATCH_FLOATING_TYPES_AND_HALF(input.scalar_type(), "grid_sampler_2d_cuda", [&] {
+      using accscalar_t = at::acc_type<scalar_t, true>;
```

Comment:
can you please use opmath_t here? Also, no need for extra template parameter, you can do `using opmath_t = ...` directly in the kernel (although it doesn't really matter, number of instantiations is still the same).